### ISSUE TITLE
Sandboxed OSX app crashes on startup

### ISSUE THREAD

**Issue by johansatge** [trueinteractions/tint2#126]
Hi,

when signed and sandboxed, my app crashes on startup on OSX Yosemite.

I was able to reproduce the issue by following the steps above *(update: you can checkout my [test repo](https://github.com/johansatge/tint-sandbox), it includes the sample app and build scripts)*

#### 1. Create a sample app

With the demo code found on the [Getting started](https://www.trueinteractions.com/tint2/docs/tutorial-gettingstarted.html) page.

App structure will look like this:

* `sample-app`
   * `index.html`
   * `main.js`
   * `package.json`
   * `someicon.png`

#### 2. Build the app

`tntbuild --out .build --clean --no-windows-build sample-app/package.json`

*The resulting `app-name-short.app` works well.*

#### 3. Sign the app

`codesign -s XXX -i com.company.app-name-short --entitlements entitlements.plist ".build/MacOS X/app-name-short.app"`

Entitlement file used:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
	<key>com.apple.security.app-sandbox</key>
	<true/>
	<key>com.apple.security.files.user-selected.read-write</key>
	<true/>
</dict>
</plist>
<!-- Is something missing ? -->
```

#### 4. Launch

It produces the error below:

```
kernel[0]: Sandbox: Runtime(25104) deny file-read-data /
kernel[0]: Sandbox: Runtime(25104) deny network-bind *:10123
com.apple.xpc.launchd[1]: (com.company.app-name-short.161344[25104]) Service exited with abnormal code: 1
```

*Is Tint sandbox-compatible ? I understood it was designed to respect the restrictions of the app stores.
BTW I like the way node and view contexts are splitted; for me it makes coding clearer than on NW.js, for instance.*

**Comment by trevorlinton** [comment]
@johansatge Tint is sandbox compatible.  We've had two people who've successfully submitted their apps to the Mac Store.  When yosemite came out we had quite a few issues with the new WKWebView  not supporting the NSURLProtocol, eventually we implemented a backup http server so that if the NSURLProtocol failed the web view could still access resources in the application.  My guess is its trying to open the http server and failing. 

This however shouldn't be much of an issue for us to fix, we've been meaning to yank the http server backup solution as Apple now supports a new version (WKURLProtocol) that implements custom schemes (e.g., the app:// scheme) in WKWebView. 

We'll prioritize this fix and get a new one out soon.  I really do appreciate you submitting these bug reports, helps us make sure Tint is meeting the highest possible standards/expectations of our users.

In the meantime (if it suits your requirements) you can add the network read/write permissions to work around the sandbox issue.  I should note that the http server only binds to the local loopback and not to any public IP address so its not creating an actual HTTP server..

At anyrate, we should hopefully have this fixed today or at max, within the next week.

**Comment by trevorlinton** [comment]
This is fixed on: https://github.com/trueinteractions/tint2/commit/d33d7b480363e445bc2c4aea9551b01d511dec99 if you still have issues, also try running ```WebView.useNewWKWebView = false``` before creating a new web view.  If you continue to have problems feel free to reopen this.  We will include this fix with v2.1.14 due out next Monday.

- Trevor

**Comment by johansatge** [comment]
Hi,

@trevorlinton thank you for this quick reply !

I compiled the code with your update and I got an odd issue with the `process.cwd()` command.

* When running with `tint`, the command returns the right path
* When built with `tntbuild`, the command returns `/`

This issue is already happening with the current version (`2.1.13`, I did not test with the previous one)

I have reproduced the error in the sample app by adding those lines:

```javascript
var exec = require('child_process').exec;
exec('syslog -s -l error "' + process.cwd() + '"');
```

**Comment by trevorlinton** [comment]
I just realized that the access violation to ```/``` shown at the beginning of this is as a sandbox error is related to packaged apps working directory being at the root.  I've fixed this in commit https://github.com/trueinteractions/tint2/commit/9e3400d442a8199aed08481cc18309f679172b13 that will be released in v2.1.14 due out tomorrow. 

Let me know if you continue to have issues.

**Comment by johansatge** [comment]
Hi,

it looks great, for now I didn't see errors when running the packaged app !

I'll keep you informed when it will have been submitted to the MAS.
